### PR TITLE
docs: Upload and view a document

### PR DESCRIPTION
**Pull request by jeromecambon**
Covers [BPM-320](https://bonitasoft.atlassian.net/browse/BPM-320)

[BPM-320]: https://bonitasoft.atlassian.net/browse/BPM-320?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by AleRondon** [comment]
Can we see an example of how to configure this document in a body for an instantiation call for example?
image

**Comment by AleRondon** [comment]
Can we please add an example on how to use the file viewer for a document with the information provided by the task context?

image

File: modules/applications/pages/ui-builder/how-to-handle-document.adoc [hunk 1]
```diff
@@ -0,0 +1,45 @@
+= Handle a document
+:description: Learn how to upload and view a document
+
+{description}
+
+* Difficulty: intermediate
+* Estimated time: 10 minutes
+
+If you want to select a document from your computer and upload it to your Bonita application, you can use the `FilePicker` widget, and the Bonita's `formFileUpload` API.
```
Comment (marcobonita) [inline modules/applications/pages/ui-builder/how-to-handle-document.adoc]:
```suggestion
If you want to select a document from your computer and upload it to your Bonita application, you can use the `FilePicker` widget along with the Bonita's `formFileUpload` API.
```

File: modules/applications/pages/ui-builder/how-to-handle-document.adoc [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+The DocumentViewer widget allows you to view the uploaded document.
```
Comment (marcobonita) [inline modules/applications/pages/ui-builder/how-to-handle-document.adoc]:
```suggestion
The `DocumentViewer` widget allows you to view the uploaded document.
```

File: modules/applications/pages/ui-builder/how-to-handle-document.adoc [hunk 3]
```diff
@@ -0,0 +1,45 @@
+= Handle a document
```
Comment (marcobonita) [inline modules/applications/pages/ui-builder/how-to-handle-document.adoc]:
```suggestion
= Upload and view documents
```

File: modules/applications/pages/ui-builder/how-to-handle-document.adoc [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+image:ui-builder/guides/file-picker-setting.png[file-picker-setting, 500]
+
+To test this step, click on the `FilePicker` widget, select a file from your computer, and click on the `Upload` button.
+
+Then, go to the `Queries` tab and check that the `uploadDocument` query has been called (`Response` tab), the status is `200 OK` and the response contains the document name.
+
+== Display the document name
+
+You can now display the document name in a text widget. Drag and drop a Text widget onto your interface and
```
Comment (marcobonita) [inline modules/applications/pages/ui-builder/how-to-handle-document.adoc]:
```suggestion
You can now display the document name in a `Text` widget. Drag and drop a `Text` widget onto your interface and
```

File: modules/applications/pages/ui-builder/how-to-handle-document.adoc [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+set the `Text` property to `{{JSON.parse(uploadDocument.data).filename}}`.
```
Comment (marcobonita) [inline modules/applications/pages/ui-builder/how-to-handle-document.adoc]:
```suggestion
set its `Text` property to `{{JSON.parse(uploadDocument.data).filename}}`.
```

File: modules/applications/pages/ui-builder/how-to-handle-document.adoc [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+Note the `JSON.parse()` call that allow to transform the response (string) into a JSON object.
```
Comment (marcobonita) [inline modules/applications/pages/ui-builder/how-to-handle-document.adoc]:
```suggestion
Note the `JSON.parse()` call, which transforms the response (string) into a JSON object.
```

**Review by marcobonita — CHANGES_REQUESTED** [review]
Typos and rephrasing

**Comment by jeromecambon** [comment]
> Can we please add an example on how to use the file viewer for a document with the information provided by the task context?
> 
> image

I succeeded to upload the document as base64, access it via a Bonita task, and view it in the DocumentViewer widget. However, the document is not viewable by UID widget (in the admin app for example), which requires a binary format.
I didn't succeed to upload the document as binary, and encode it as base64 to make it readable by the widget. I encoded as base64, but I still have a blank result on the widget...
I propose to merge this PR, since it add some interesting values. We can handle the issue separately.

**Review by marcobonita — APPROVED** [review]

**Review by benjaminParisel — APPROVED** [review]

**Pull request merged**